crates/core_types/src/network/entry_aspect.rs

```diff
@@ -95,12 +97,24 @@ impl EntryAspect {
     }
     /// NB: this is the inverse function of entry_to_meta_aspect,
     /// so it is very important that they agree!
-    pub fn entry_address(&self) -> &Address {
-        match self {
-            EntryAspect::LinkAdd(link_data, _) => link_data.link.base(),
-            EntryAspect::LinkRemove((link_data, _), _) => link_data.link.base(),
-            _ => self.header().entry_address(),
-        }
+    pub fn entry_address(&self) -> Result<Address, HolochainError> {
```

Comment:
and here is the aspect -> entry map, which the inverse map now uses to maintain agreement between the two